src/main/java/nextstep/auth/authentication/SessionAuthenticationConverter.java

```diff
@@ -0,0 +1,18 @@
+package nextstep.auth.authentication;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+public class SessionAuthenticationConverter implements AuthenticationConverter {
+    public static final String USERNAME_FIELD = "username";
+    public static final String PASSWORD_FIELD = "password";
+
+    @Override
+    public AuthenticationToken convert(HttpServletRequest request) {
+        Map<String, String[]> paramMap = request.getParameterMap();
+        String principal = paramMap.get(USERNAME_FIELD)[0];
```

Comment:
배열 인덱스로 직접 지정해주시는 것보다 Stream으로 감싸서 없을 땐 4xx 에러를 발생시키는게 좋겠네요.

```suggestion
        String principal = Stream.of(request.getParameterValues(USERNAME_FIELD))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("적절한 에러 메시지"));
```

Reply:
오👍🏼 저렇게 빼낼수도 있군요.. 적용했습니다 감사합니다!